src/Components/Games/EmbeddedLink.svelte

```diff
@@ -1,32 +1,88 @@
-    type EmbeddedType = "png" | "gif" | "mp4";
+    import Youtube from "svelte-youtube";
+    type EmbeddedType = "png" | "gif" | "mp4" | "webm" | "youtube" | "tiktok";
 
     export let embeddedLink: string;
     export let aspectRatio: "expand" | "preserveRatio" = "preserveRatio";
-    // TODO: Parse link to deduce type: png / gif / mp4 etc etc etc etc
 
+    let videoID: string = null;
     let embeddedType: EmbeddedType = getEmbeddedType(embeddedLink);
 
     function getEmbeddedType(link: string): EmbeddedType {
+        if (embeddedLink.startsWith("https://www.youtube.com") || embeddedLink.startsWith("www.youtube.com")) {
+            videoID = getYoutubeId(link);
+            if (videoID === null) {
+                console.error("TYPE " + link + " Isn't recognized.");
+                return;
+            }
+            return "youtube";
+        }
+        if (embeddedLink.startsWith("https://www.tiktok.com") || embeddedLink.startsWith("www.tiktok.com")) {
+            videoID = embeddedLink.split("/")[embeddedLink.split("/").length - 1];
+            return "tiktok";
+        }
+
         if (embeddedLink.endsWith(".png")) return "png";
         else if (embeddedLink.endsWith(".gif")) return "gif";
         else if (embeddedLink.endsWith(".mp4")) return "mp4";
+        else if (embeddedLink.endsWith(".webm")) return "webm";
+
         console.error("TYPE " + link + " Isn't recognized.");
     }
+
+    function getYoutubeId(link: string): string | null {
+        var regExp = /^.*((youtu.be\/)|(v\/)|(\/u\/\w\/)|(embed\/)|(watch\?))\??v?=?([^#&?]*).*/;
+        var match = link.match(regExp);
+        return match && match[7].length == 11 ? match[7] : null;
+    }
+
+    const YoutubeOptions = {
+        width: "100%",
+        height: "100%",
+        playerVars: {
+            rel: 0,
+            modestbranding: 1,
+            autoplay: 1,
+        },
+    };
+
+    function onYTVideoEnd(event) {
+        event.detail.target.playVideo();
+    }
 </script>
 
 {#if embeddedType === "png"}
     <img src={embeddedLink} alt="Embedded link" class={aspectRatio} />
 {:else if embeddedType === "gif"}
-    <div />
-{:else if embeddedType === "mp4"}
-    <div />
+    <div class="expand ">
+        <img src={embeddedLink} alt="Embedded link" class={aspectRatio} />
+    </div>
+{:else if embeddedType === "mp4" || embeddedType === "webm"}
+    <div class="expand blackBackground">
+        <video src={embeddedLink} class={aspectRatio} autoplay loop controls><track kind="captions" /></video>
+    </div>
+{:else if embeddedType === "youtube"}
+    <Youtube videoId={videoID} options={YoutubeOptions} class="YOUTUBE_VIDEO_CONTAINER" on:end={onYTVideoEnd} />
+{:else if embeddedType === "tiktok"}
+    <!-- <blockquote
+        class="tiktok-embed"
+        cite="https://www.tiktok.com/@tceladev/video/{videoID}"
+        data-video-id={videoID}
+        style="width: 500px; height: 500px; scroll:hidden;"
+    >
+        <section />
+    </blockquote> -->
```

Comment:
This is the default way that tiktok recomends to embedd they videos. (I dont know why, but it doesn't always charge de embeded iframe).
Might be useful to investigate more.